### DISCUSSION TITLE
Gratisdata and good linked data

### DISCUSSION THREAD

**Discussion by ugochimobi** [schemaorg/schemaorg#3089] [General]
This discussion is sequel to #3049, a ton of discussions were had there and this is a continuation. 
My last question/comment there was 
@thadguidry  While going through the documentations about enabling this feature for Gratisdata, I couldn't find any information regarding making it possible to have multiple sameAs properties. I know it's possible, but how can that be accomplished? (I'm sorry if I'm sounding dumb as I'm relatively new in the linked data network generally).

How did you get the working example so fast?

Anyone can help please.

**Comment by thadguidry** [comment]
P2888 [exact match](https://www.wikidata.org/wiki/Property:P2888) is an equivalent property to http://schema.org/sameAs and also http://www.w3.org/2004/02/skos/core#exactMatch
It was one of the first properties that we mapped in Wikidata in order to fulfill further mappings in the linked data ecosystem.

Anyways, in your case, in the Gratisdata Wikibase instance, you will need to find out how to enable that property availability, because I don't see it:
![image](https://user-images.githubusercontent.com/986438/163232853-8adca126-d66a-4803-8675-3a75f2e26dac.png)

and then after fixing that you can make multiple `exact match` statements with each of my example URL's that I gave you like so:
```json
"sameAs": ["https://stias.ac.za/fellows/eghosa-emmanuel-osaghae/", "https://niia.gov.ng/about-us/dg/", "https://gratisdata.miraheze.org/wiki/Q2955", "https://www.strausscenter.org/person/eghosa-osaghae-2/"]
```

**Reply by thadguidry** [reply]
Finally, you can then use and pull in your Gratisdata Wikibase data "inside" your Gratispaideia article of the same entity https://gratispaideia.miraheze.org/wiki/Eghosa_Emmanuel_Osaghae
Effectively having `sameAs` or rather `exact match` statements inside your Gratispaideia article.  Just make sure whatever template or extension system you are using in your Mediawiki instance that it is providing "good JSON-LD" or "decent RDFa or microdata structure".  How to do **that** would be best asked to the Mediawiki/Wikipedia public mailing list.  (because I haven't kept up with the Mediawiki template/extensions changes over the last 3 years)

**Reply by ugochimobi** [reply]
> P2888 [exact match](https://www.wikidata.org/wiki/Property:P2888) is an equivalent property to http://schema.org/sameAs and also http://www.w3.org/2004/02/skos/core#exactMatch It was one of the first properties that we mapped in Wikidata in order to fulfill further mappings in the linked data ecosystem.
> 
> Anyways, in your case, in the Gratisdata Wikibase instance, you will need to find out how to enable that property availability, because I don't see it: ![image](https://user-images.githubusercontent.com/986438/163232853-8adca126-d66a-4803-8675-3a75f2e26dac.png)
> 
> and then after fixing that you can make multiple `exact match` statements with each of my example URL's that I gave you like so:
> 
> ```json
> "sameAs": ["https://stias.ac.za/fellows/eghosa-emmanuel-osaghae/", "https://niia.gov.ng/about-us/dg/", "https://gratisdata.miraheze.org/wiki/Q2955", "https://www.strausscenter.org/person/eghosa-osaghae-2/"]
> ```

@thadguidry I'm very sorry for the inconveniences, there's currently an issue with that configurations and we're currently fixing. But that being said, the `exact match ` property is P2888 in wikidata, but it's not on gratisdata, it is `P34` on gratisdata.

**Reply by ugochimobi** [reply]
> Finally, you can then use and pull in your Gratisdata Wikibase data "inside" your Gratispaideia article of the same entity https://gratispaideia.miraheze.org/wiki/Eghosa_Emmanuel_Osaghae Effectively having `sameAs` or rather `exact match` statements inside your Gratispaideia article. Just make sure whatever template or extension system you are using in your Mediawiki instance that it is providing "good JSON-LD" or "decent RDFa or microdata structure". How to do **that** would be best asked to the Mediawiki/Wikipedia public mailing list. (because I haven't kept up with the Mediawiki template/extensions changes over the last 3 years)

I would ask at mediawiki wikibase mailing list, but can i CC you? how do I cc you?

**Reply by thadguidry** [reply]
Please don't cc me. Thanks.

**Comment by ugochimobi** [comment]
Okay, thanks!

**Comment by ugochimobi** [comment]
@thadguidry Hi, I hope you're having a great day.
Please regarding the `exact match` property, Gratisdata's own is [P34](https://gratisdata.miraheze.org/wiki/Property:P34 )and so it has not been mapped with Schema.org's `sameAs` property yet... Please can that be done?

We don't use Wikibase Federation on Gratisdata so we literally can't use Wikidata's properties...

**Reply by thadguidry** [reply]
@ugochimobi You cannot use Wikidata's properties, understood... but you can map to Wikidata's properties.
You could even store that mapping inside Wikidata's `exact match` property (or store it in Gratisdata's) with an equivalent property statement added to it, as I have done previously for Schema.org and SKOS here:
https://www.wikidata.org/wiki/Property:P2888
You can do this yourself.  Also, it's best to map to the Concept URL https://gratisdata.miraheze.org/entity/P34 .  (notice there is no `/wiki/`).  This will help with machine-readable structured linked data.

**Reply by ugochimobi** [reply]
@thadguidry Ah, thank you very much for your comprehensive reply, it went a long way for me.

> You could even store that mapping inside Wikidata's exact match property

Are you saying, I should request on Wikidata, that Gratisdata's P34 be added as an exact match value for Wikidata's?

Or on the other hand,
> or store it in Gratisdata's 

I should do the reverse of above on Gratisdata?

You know, what I was thinking is that Schema.org currently understands Wikidata's properties but not Gratisdata, so that's why I don't think the second option might work as we want it, although I might not be correct in that.

**Reply by thadguidry** [reply]
To be clear, Schema.org is a standard for structured data... it has no brain to derive understanding.  Search engines, consuming apps, and data programs use Schema.org's vocabulary standards, syntax, and structure, to derive understanding to help machines understand... which helps humans more.

You do not have to **request** adding something to Wikidata... it's a Wiki ! ... you are allowed to do it yourself (just create an good account or reuse the one you might already have for Wikipedia) and add the `equivalent property` statement with your Concept URL.

Schema.org currently does not store mappings, or provide the web of linked data... that is the responsibility of Wikidata, consuming apps, other Linked Data sites, aggregators, and Search engines. (including Gratisdata)

On the reverse side... You are certainly in control of Gratisdata, so you can **also** store mappings within it however you wish.

**Reply by ugochimobi** [reply]
> To be clear, Schema.org is a standard for structured data... it has no brain to derive understanding. Search engines, consuming apps, and data programs use Schema.org's vocabulary standards, syntax, and structure, to derive understanding to help machines understand... which helps humans more.
> 
> You do not have to **request** adding something to Wikidata... it's a Wiki ! ... you are allowed to do it yourself (just create an good account or reuse the one you might already have for Wikipedia) and add the `equivalent property` statement with your Concept URL.
> 
> Schema.org currently does not store mappings, or provide the web of linked data... that is the responsibility of Wikidata, consuming apps, other Linked Data sites, aggregators, and Search engines. (including Gratisdata)
> 
> On the reverse side... You are certainly in control of Gratisdata, so you can **also** store mappings within it however you wish.

@thadguidry Oh oh, actually the reason I used to word "request" was that you know, some properties might be protected from editing, and it depends on the protection level used, so if the case is it's above my own account level, then I can't edit it directly, that is why I used that word.

And thank you very much, I actually understand that it's a Wiki, and I know what that means too. Thank you for the explanation too, very much appreciated!

**Comment by ugochimobi** [comment]
@thadguidry Please take a look at https://gratisdata.miraheze.org/wiki/Gratisdata:Contact_the_development_team#Find_a_way_to_map_our_property_concept_URIs_with_Wikidata
I just want to know how accurate I am.
Thanks and hope to hear from you soon.

**Reply by thadguidry** [reply]
Perfect.

**Reply by thadguidry** [reply]
Also of note:  SKOS relations are also available in Wikidata... so, if there is a Broader/Narrower property and it's not `exact match` or sameAs then you could use those SKOS relations.  I've done a few of them here and there inside Wikidata as I went along with the Schema.org mapping project so take a look.  I think I mentioned a few inside the #280 issue, but cannot recall.

**Reply by thadguidry** [reply]
Here's the WDQS link to show the current ontology mapping properties that are available inside Wikidata thus far, where you could query to see if any entities have some of those properties attached: 
https://query.wikidata.org/#SELECT%20DISTINCT%20%3Fitem%20%3FitemLabel%20WHERE%20%7B%0A%20%20SERVICE%20wikibase%3Alabel%20%7B%20bd%3AserviceParam%20wikibase%3Alanguage%20%22%5BAUTO_LANGUAGE%5D%22.%20%7D%0A%20%20%7B%0A%20%20%20%20SELECT%20DISTINCT%20%3Fitem%20WHERE%20%7B%0A%20%20%20%20%20%20%3Fitem%20p%3AP31%20%3Fstatement0.%0A%20%20%20%20%20%20%3Fstatement0%20%28ps%3AP31%2F%28wdt%3AP279%2a%29%29%20wd%3AQ30249126.%0A%20%20%20%20%7D%0A%20%20%20%20LIMIT%20100%0A%20%20%7D%0A%7D

For further help on writing SPARQL, ask here https://www.wikidata.org/wiki/Wikidata:Request_a_query

**Reply by ugochimobi** [reply]
> Also of note: SKOS relations are also available in Wikidata... so, if there is a Broader/Narrower property and it's not `exact match` or sameAs then you could use those SKOS relations. I've done a few of them here and there inside Wikidata as I went along with the Schema.org mapping project so take a look. I think I mentioned a few inside the #280 issue, but cannot recall.

Please can you briefly explain what you mean by "a Broader/Narrower property" and also what you mean by "it's not `exact match` or sameAs"

**Reply by ugochimobi** [reply]
> 

Thanks for the links, much appreciated!
I'd take a look.

**Reply by thadguidry** [reply]
I think it would be better if you can educate yourself more by reading some of the SKOS documentation and also the SKOS Primer (link is in it's Abstract) https://www.w3.org/TR/skos-reference/  and within it... specifically Section 8 SEMANTIC RELATIONS will answer your question directly. :-)

**Reply by ugochimobi** [reply]
> I think it would be better if you can educate yourself more by reading some of the SKOS documentation and also the SKOS Primer (link is in it's Abstract) https://www.w3.org/TR/skos-reference/ and within it... specifically Section 8 SEMANTIC RELATIONS will answer your question directly. :-)

Alright, I'd do so, let me have a look.

**Reply by ugochimobi** [reply]
And thank you for the amendments you made to the Gratisdata Dev page I linked above ;)

**Comment by ugochimobi** [comment]
@thadguidry Good evening and Happy New Week, I hope you're having a great time and doing good as well.
Just wanted to ask; Just as we want to map Gratisdata's properties to Wikidata properties with the `equivalent property` property, is it also sensible to say that we can also map Gratisdata's items to Wikidata's items with the `equivalent class` property?

and I was also thinking since this mapping process is or might be very large in terms of editing and may even require other volunteers to help out, I was thinking that maybe we might need to formally propose (or maybe that's not the best word to use) on Wikidata so that it won't look like we're spamming Wikidata's entities, because similar things have happened in the past and they ended up in Wikidata's MediaWiki:Spam-blacklist.
What do you think?

**Reply by thadguidry** [reply]
> @thadguidry Good evening and Happy New Week, I hope you're having a great time and doing good as well. Just wanted to ask; Just as we want to map Gratisdata's properties to Wikidata properties with the `equivalent property` property, is it also sensible to say that we can also map Gratisdata's items to Wikidata's items with the `equivalent class` property?
> 

Yes.  On Wikidata's side you can add a P1709 statement with your Gratisdata URI for the concept or entity, whichever.

> and I was also thinking since this mapping process is or might be very large in terms of editing and may even require other volunteers to help out, I was thinking that maybe we might need to formally propose (or maybe that's not the best word to use) on Wikidata so that it won't look like we're spamming Wikidata's entities, because similar things have happened in the past and they ended up in Wikidata's MediaWiki:Spam-blacklist. What do you think?

Would not be considered spam.  I would definitely vote for it as a furthering of Wikidata's (Wikipedia's) mission.  One of which is allowing more federated linked data especially for special domains, etc. rather than Wikidata continuing to absorb, store & grow values that might not be suited for a general KOS.

**Reply by ugochimobi** [reply]
> Yes. On Wikidata's side you can add a P1709 statement with your Gratisdata URI for the concept or entity, whichever.

Okay, thank you very much for this

> Would not be considered spam. I would definitely vote for it as a furthering of Wikidata's (Wikipedia's) mission. One of which is allowing more federated linked data especially for special domains, etc. rather than Wikidata continuing to absorb, store & grow values that might not be suited for a general KOS.

This sounds good and thank you! 
So, I was also wondering, for those properties that are protected, like I said earlier, could you or any other Wikidatan who has sufficient permissions do the mapping for us? 
Because I've seen a ton of them that I can't edit (Although I've spent over 1yr on Wikimedia).

**Reply by thadguidry** [reply]
For those properties that are protected from edits, you can open a new topic on any property's Talk/Discussion page and give a paragraph ending with your signature.  Alternatively, you can compose an email, including the list and value changes needed of those properties and send to the [Wikidata mailing list](https://www.wikidata.org/wiki/Wikidata:Mailing_list)

**Reply by ugochimobi** [reply]
> For those properties that are protected from edits, you can open a new topic on any property's Talk/Discussion page and give a paragraph ending with your signature. Alternatively, you can compose an email, including the list and value changes needed of those properties and send to the [Wikidata mailing list](https://www.wikidata.org/wiki/Wikidata:Mailing_list)

Okay, thank you very much.

**Comment by ugochimobi** [comment]
Hi, @thadguidry Good evening and I hope you're doing great!
I wanted to let you know of my progress in the mapping on that same page on Gratisdata, [here](https://gratisdata.miraheze.org/wiki/Gratisdata:Contact_the_development_team#Gratisdata_props_to_be_mapped_with_Wikidata%27s_props) \o/
I also want to let you know that, I was able to edit properties immediately I became autoconfirmed on wikidata (50 edits + a 4-day-old account) which I have actually passed the account age. \o/

Also, I want to ask; How do I know that the mappings are working, like is there a way to know or something?

Hope to hear from you soon... thanks!

**Comment by thadguidry** [comment]
How do you want the mappings to work? :-)  What is your expectation?

**Reply by ugochimobi** [reply]
> How do you want the mappings to work? :-) What is your expectation?

@thadguidry I sincerely don't know how to know, I don't even know what I expect, lol. How do I know, please?

**Reply by thadguidry** [reply]
So consider 2 databases.  Both databases share a common field/column in one of their tables.  Let's call that field "PlantFamilyName".  You could then query across both databases via UNION's and JOIN's to see the results of additional commonality or uniqueness in one database compared to the other.  And essentially be able to tell what are differences and what is shared between the databases.

Having a map, allows one to already know that 2 particular fields/columns represent data considered to be of the "same Type of Thing", as just one example of the usefulness of a map.

**Reply by ugochimobi** [reply]
@thadguidry If I tell you that I understood any of the above, you'd know that I'm lieing ◉‿◉

When you're talking about databases, were you insinuating something like a Neo4j database?

**Reply by thadguidry** [reply]
Neo4j is a property graph database.  I was referring to RDBMS database systems (traditional tables) that can be queried with SQL.

**Reply by ugochimobi** [reply]
> Neo4j is a property graph database. I was referring to RDBMS database systems (traditional tables) that can be queried with SQL.

Oh, I read about RDBMS and I saw that, although I'm still not sure what I should do -_-

**Comment by ugochimobi** [comment]
Hi @thadguidry Good morning and thank you for your assistance so far, I hope you're doing great.
I want to link you to a thread that was dropped on my talk page, https://www.wikidata.org/wiki/User_talk:Ugochimobi#gratisdata.miraheze.org_spam
The user claimed it was spam, please can you see to that?

thank you and I'd appreciate

**Reply by thadguidry** [reply]
@ugochimobi I have taken a look at your excellent response to them and also included my own.  This should be sorted out shortly internally very soon.  Sit back and relax for a while.

**Comment by ugochimobi** [comment]
Okay, thank you very much, I appreciate and remain calm and see what
happens.

**Comment by ugochimobi** [comment]
Hi, @thadguidry Good evening and I hope you're having a great time over there :)
What then do we do about this at this point? because I never thought Wikidata would be the best solution :(

**Reply by thadguidry** [reply]
Solution for what?

**Comment by ugochimobi** [comment]
@thadguidry this mapping of a thing
